src/compiler/program.ts

```diff
@@ -1031,8 +1033,10 @@ namespace ts {
                     diagnosticArgument = [fileName, "'" + supportedExtensions.join("', '") + "'"];
                 }
                 else if (!findSourceFile(fileName, toPath(fileName, currentDirectory, getCanonicalFileName), isDefaultLib, refFile, refPos, refEnd)) {
-                    diagnostic = Diagnostics.File_0_not_found;
-                    diagnosticArgument = [fileName];
+                    if (hasTypeScriptFileExtension(fileName) || options.disableSizeLimit || programSizeForNonTsFiles !== -1) {

```

Comment:
This magic `-1` value is being used in a few places now. Can you make this a `const` with a meaningful name and use that to make the code more readable (e.g. `const sizeLimitExceeded = -1`). It would also help readability if this was a simple helper function (e.g. `function sizeLimitExceed() { return !options.disableLimit && programSizeForNonTsFiles === sizeLimitExceeded;}`) used in these conditions.